RecoEgamma/EgammaTools/plugins/EGExtraInfoModifierFromDB.cc

```diff
@@ -371,102 +381,109 @@ namespace {
-  assignValue(ptr, e_conf.tag_float_token_map.find(std::string("clusterDPhiToSeed2"))->second.second, ele_vmaps, clusterDPhiToSeed2); 
-  assignValue(ptr, e_conf.tag_float_token_map.find(std::string("clusterDEtaToSeed0"))->second.second, ele_vmaps, clusterDEtaToSeed0); 
-  assignValue(ptr, e_conf.tag_float_token_map.find(std::string("clusterDEtaToSeed1"))->second.second, ele_vmaps, clusterDEtaToSeed1); 
-  assignValue(ptr, e_conf.tag_float_token_map.find(std::string("clusterDEtaToSeed2"))->second.second, ele_vmaps, clusterDEtaToSeed2); 
-  assignValue(ptr, e_conf.tag_int_token_map.find(std::string("iPhi"))->second.second, ele_int_vmaps, iPhi);
-  assignValue(ptr, e_conf.tag_int_token_map.find(std::string("iEta"))->second.second, ele_int_vmaps, iEta);
-  assignValue(ptr, e_conf.tag_float_token_map.find(std::string("cryPhi"))->second.second, ele_vmaps, cryPhi);
-  assignValue(ptr, e_conf.tag_float_token_map.find(std::string("cryEta"))->second.second, ele_vmaps, cryEta);
-
-  eval[8]  = eMax/sc->rawEnergy();
-  eval[9]  = e2nd/sc->rawEnergy();
-  eval[10] = (eLeft+eRight!=0. ? (eLeft-eRight)/(eLeft+eRight) : 0.);
-  eval[11] = (eTop+eBottom!=0. ? (eTop-eBottom)/(eTop+eBottom) : 0.);
+  eval[7]  = theseed->energy()/raw_energy;
+  eval[8]  = ess.eMax/raw_energy;
+  eval[9]  = ess.e2nd/raw_energy;
+  eval[10] = (ess.eLeft + ess.eRight != 0.f  ? (ess.eLeft-ess.eRight)/(ess.eLeft+ess.eRight) : 0.f);
+  eval[11] = (ess.eTop  + ess.eBottom != 0.f ? (ess.eTop-ess.eBottom)/(ess.eTop+ess.eBottom) : 0.f);
   eval[12] = ele.sigmaIetaIeta();
-  eval[13] = sieip;
+  eval[13] = ess.sigmaIetaIphi;
   eval[14] = ele.sigmaIphiIphi();
-  const int N_ECAL = sc->clustersEnd() - sc->clustersBegin();
-  eval[15] = std::max(0,N_ECAL - 1);
+  eval[15] = std::max(0,numberOfClusters-1);
+  
+  // calculate sub-cluster variables
+  std::vector<float> clusterRawEnergy;
+  clusterRawEnergy.resize(std::max(3, numberOfClusters), 0);
+  std::vector<float> clusterDEtaToSeed;
+  clusterDEtaToSeed.resize(std::max(3, numberOfClusters), 0);
+  std::vector<float> clusterDPhiToSeed;
+  clusterDPhiToSeed.resize(std::max(3, numberOfClusters), 0);
+  float clusterMaxDR     = 999.;
+  float clusterMaxDRDPhi = 999.;
+  float clusterMaxDRDEta = 999.;
+  float clusterMaxDRRawEnergy = 0.;
+  
+  size_t iclus = 0;
+  float maxDR = 0;
+  edm::Ptr<reco::CaloCluster> pclus;
+  // loop over all clusters that aren't the seed  
+  auto clusend = the_sc->clustersEnd();
+  for( auto clus = the_sc->clustersBegin(); clus != clusend; ++clus ) {
+    pclus = *clus;
+    
+    if(theseed == pclus ) 
+      continue;
+    clusterRawEnergy[iclus]  = pclus->energy();
+    clusterDPhiToSeed[iclus] = reco::deltaPhi(pclus->phi(),theseed->phi());
+    clusterDEtaToSeed[iclus] = pclus->eta() - theseed->eta();
+    
+    // find cluster with max dR
+    if(reco::deltaR(*pclus, *theseed) > maxDR) {
```

Comment:
better not copy-paste the function call  (even though the optimized code will take it out most likely)